gap/attr.gi

```diff
@@ -1336,6 +1336,42 @@ end);
 # It is the backend to IsBipartiteDigraph, Bicomponents, and DigraphColouring
 # for a 2-colouring
 
+InstallMethod(DigraphMycielskian, "for a symmetric digraph",
```

Comment:
I think we really need a different version of this for mutable and immutable digraphs; the one for immutable digraphs would look like this:
```
if not IsSymmetricDigraph(D) or IsMultiDigraph(D) then
  (blah)
fi;
return MakeImmutable(DigraphMycielskian(DigraphMutableCopy(D)));
```
And the mutable version would be the version that exists at the moment.

As things stand at the moment, for an immutable digraph, the current function makes a new copy in memory of the existing digraph every time you add a vertex or an edge, which happens a large number of times and harms performance.